src/query/storage/m3/storage.go

```diff
@@ -711,28 +711,14 @@ func (s *m3storage) Write(
 		// capture var
 		datapoint := datapoint
 		wg.Add(1)
-
-		var (
-			now                      = time.Now()
-			deadline, deadlineExists = ctx.Deadline()
-			timeout                  = minWriteWaitTimeout
-		)
-		if deadlineExists {
-			if remain := deadline.Sub(now); remain >= timeout {
-				timeout = remain
-			}
-		}
-		spawned := s.opts.WriteWorkerPool().GoWithTimeout(func() {
+		s.opts.WriteWorkerPool().AlwaysGo(func() {
```

Comment:
nit (and not your change): Maybe we should pull this off of the options in the constructor for m3storage? Seems strange we access options each time for this.